File: StudioZaneeMusic(Mobile)/lib/models/recording_booking.dart
class RecordingBooking {
  int? id;
  String customerName;
  String customerEmail;
  String customerPhone;
  String recordingPackage;
  int? price;
  String recordingDate; // yyyy-MM-dd
  String recordingTime; // HH:mm:ss
  int duration;
  String? status;

  RecordingBooking({
    this.id,
    required this.customerName,
    required this.customerEmail,
    required this.customerPhone,
    required this.recordingPackage,
    this.price,
    required this.recordingDate,
    required this.recordingTime,
    required this.duration,
    this.status,
  });

  Map<String, dynamic> toJson() => {
    'customerName': customerName,
    'customerEmail': customerEmail,
    'customerPhone': customerPhone,
    'recordingPackage': recordingPackage,
    'price': price ?? 0,
    'recordingDate': recordingDate,
    'recordingTime': recordingTime,
    'duration': duration,
    'status': status ?? "pending",
  };

  factory RecordingBooking.fromJson(Map<String, dynamic> json) =>
      RecordingBooking(
        id: json['id'],
        customerName: json['customerName'],
        customerEmail: json['customerEmail'],
        customerPhone: json['customerPhone'],
        recordingPackage: json['recordingPackage'],
        price: json['price'],
        recordingDate: json['recordingDate'],
        recordingTime: json['recordingTime'],
        duration: json['duration'],
        status: json['status'],
      );
}

File: StudioZaneeMusic(Web)/Models/RecordingBooking.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace DuAnBai3.Models
{
    public class RecordingBooking
    {
        public int Id { get; set; }

        [Required]
        public string CustomerName { get; set; }

        [Required, EmailAddress]
        public string CustomerEmail { get; set; }

        [Required]
        public string CustomerPhone { get; set; }

        [Required]
        public string RecordingPackage { get; set; }

        public int Price { get; set; }

        public DateTime RecordingDate { get; set; }

        public TimeSpan RecordingTime { get; set; }

        [Range(1, 8)]
        public int Duration { get; set; } = 1; // Thời lượng thu âm (giờ)

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public string Status { get; set; } = "pending";
    }
}
